pyuvdata/uvcal/uvcal.py

```diff
@@ -3200,6 +3248,58 @@ def __add__(
                             np.ix_(ants_t2o, [0], freqs_t2o, times_t2o, jones_t2o)
                         ] = other.input_flag_array
 
+            # Fix ordering
+            ant_axis_num = 0
+            if this.future_array_shapes:
+                faxis_num = 1
+                taxis_num = 2
+                jaxis_num = 3
+            else:
+                faxis_num = 2
+                taxis_num = 3
+                jaxis_num = 4
+
+            axis_dict = {
+                ant_axis_num: {"inds": anew_inds, "order": ant_order},
+                faxis_num: {"inds": fnew_inds, "order": f_order},
+                taxis_num: {"inds": tnew_inds, "order": t_order},
+                jaxis_num: {"inds": jnew_inds, "order": j_order},
+            }
+
+            for name, param in zip(this._data_params, this.data_like_parameters):
+                if param is None:
+                    continue
+                axis_delta = 0
+                if name == "total_quality_array":
+                    axis_delta = 1
+
+                for axis, subdict in axis_dict.items():
+                    if len(subdict["inds"]) > 0:
+                        setattr(
+                            this,
+                            name,
+                            np.take(param, subdict["order"], axis=axis - axis_delta),
```

Comment:
Always good to see commits where the subtractions are twice the number of lines of additions 😉